$NetBSD: patch-configure,v 1.1 2023/09/13 07:36:44 taca Exp $

* More support for OpenSSL 1.1; not only check SSL_Library_init() but
  also check OPENSSL_init_ssl().
* Fix syntax error by accidental new line.
* Utilize <stdlib.h> on BSD.
* Do not override CFLAGS/CXXFLAGS except linux.
* Prevent WRKDIR reference.

--- configure.orig	2023-07-06 05:28:38.000000000 +0000
+++ configure
@@ -1116,9 +1116,6 @@ CXXFLAGS
 CCC
 BUILDCXX
 CPP
-PKG_CONFIG
-PKG_CONFIG_PATH
-PKG_CONFIG_LIBDIR
 LT_SYS_LIBRARY_PATH
 CXXCPP
 LIBEXPAT_CFLAGS
@@ -29233,10 +29230,13 @@ do
 done
 
   # GLIBC 2.30 deprecates sysctl.h. Test with the same flags that (may) break includes later.
-  CFLAGS=$SQUID_CFLAGS
-  CXXFLAGS=$SQUID_CXXFLAGS
   case "$squid_host_os" in #(
-  linux|solaris|freebsd|openbsd|netbsd|cygwin) :
+  linux) :
+    CFLAGS=$SQUID_CFLAGS
+    CXXFLAGS=$SQUID_CXXFLAGS
+    : ;; #(
+  solaris|freebsd|openbsd|netbsd|cygwin) :
+    ${TRUE}
     : ;; #(
   mingw) :
     EUILIB="-liphlpapi" ;; #(
@@ -29284,6 +29284,7 @@ then :
 fi
 
   ac_fn_cxx_check_header_compile "$LINENO" "net/if_arp.h" "ac_cv_header_net_if_arp_h" "
+#include <stdio.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 
@@ -29294,6 +29295,7 @@ then :
 
 fi
 ac_fn_cxx_check_header_compile "$LINENO" "net/route.h" "ac_cv_header_net_route_h" "
+#include <stdio.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 
@@ -30488,7 +30490,51 @@ then :
   LIBOPENSSL_LIBS="-lssl $LIBOPENSSL_LIBS"
 else $as_nop
 
-      as_fn_error $? "library 'ssl' is required for OpenSSL" "$LINENO" 5
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for OPENSSL_init_ssl in -lssl" >&5
+$as_echo_n "checking for OPENSSL_init_ssl in -lssl... " >&6; }
+if ${ac_cv_lib_ssl_OPENSSL_init_ssl+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lssl $LIBOPENSSL_LIBS $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char OPENSSL_init_ssl ();
+int
+main ()
+{
+return OPENSSL_init_ssl ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_cxx_try_link "$LINENO"; then :
+  ac_cv_lib_ssl_OPENSSL_init_ssl=yes
+else
+  ac_cv_lib_ssl_OPENSSL_init_ssl=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_ssl_OPENSSL_init_ssl" >&5
+$as_echo "$ac_cv_lib_ssl_OPENSSL_init_ssl" >&6; }
+if test "x$ac_cv_lib_ssl_OPENSSL_init_ssl" = xyes; then :
+  LIBOPENSSL_LIBS="-lssl $LIBOPENSSL_LIBS"
+else
+
+	as_fn_error $? "library 'ssl' is required for OpenSSL" "$LINENO" 5
+
+fi
+
+
 
 fi
 
@@ -30592,7 +30638,51 @@ then :
   LIBOPENSSL_LIBS="-lssl $LIBOPENSSL_LIBS"
 else $as_nop
 
-      as_fn_error $? "library 'ssl' is required for OpenSSL" "$LINENO" 5
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for OPENSSL_init_ssl in -lssl" >&5
+$as_echo_n "checking for OPENSSL_init_ssl in -lssl... " >&6; }
+if ${ac_cv_lib_ssl_OPENSSL_init_ssl+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lssl $LIBOPENSSL_LIBS $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char OPENSSL_init_ssl ();
+int
+main ()
+{
+return OPENSSL_init_ssl ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_cxx_try_link "$LINENO"; then :
+  ac_cv_lib_ssl_OPENSSL_init_ssl=yes
+else
+  ac_cv_lib_ssl_OPENSSL_init_ssl=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_ssl_OPENSSL_init_ssl" >&5
+$as_echo "$ac_cv_lib_ssl_OPENSSL_init_ssl" >&6; }
+if test "x$ac_cv_lib_ssl_OPENSSL_init_ssl" = xyes; then :
+  LIBOPENSSL_LIBS="-lssl $LIBOPENSSL_LIBS"
+else
+
+	as_fn_error $? "library 'ssl' is required for OpenSSL" "$LINENO" 5
+
+fi
+
+
 
 fi
 
@@ -51080,6 +51170,8 @@ else
 # elif defined _MSC_VER
 #  include <malloc.h>
 #  define alloca _alloca
+# elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__OpenBSD__)
+#   include <stdlib.h>
 # else
 #  ifdef  __cplusplus
 extern "C"
@@ -54207,6 +54299,9 @@ printf "%s\n" "#define USE_SOLARIS_IPFIL
 #if USE_SOLARIS_IPFILTER_MINOR_T_HACK
 #define minor_t fubar
 #endif
+#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
